firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -72,16 +72,23 @@ class BrowserViewController: UIViewController,
     weak var navigationHandler: BrowserNavigationHandler?
 
     private(set) lazy var addressToolbarContainer: AddressToolbarContainer = .build()
-    var urlBar: URLBarView!
+    var legacyUrlBar: URLBarView?
 
-    var urlBarHeightConstraint: Constraint!
+    var urlBarView: (URLBarViewProtocol & TopBottomInterchangeable & Autocompletable) {
+        if !isToolbarRefactorEnabled, let legacyUrlBar {
+            return legacyUrlBar
+        }
+        return addressToolbarContainer
+    }
```

Comment:
There was a lot of ternary statements returning either `urlBar` or `addressToolbarContainer` based on the value of `isToolbarRefactorEnabled` in this view controller.

To simplify things, my approach is to add this new computed property that will check for the state of `isToolbarRefactorEnabled` but also safely unwrap `urlBar`.

I've also took the liberty to rename `urlBar` to `legacyUrlBar` for the sake of clarity (since the method creating it was already called `createLegacyUrlBar()` and there is already a function named `urlBar` so it also avoids confusion).

Let me know what you think 🙂 The rest of this PR's changes are mostly subsequent to this change.

Reply:
Hi @tonell-m, I just wanted to say thank you very much for explaining your reasoning, it makes reviewing easier to have the extra context and a glimpse into your thought process. 👌 

Excellent work! You definitely made the right calls here; this code is much cleaner now. Thanks for your effort on these tickets! 🔥

Reply:
Thanks a lot! Pleasure is mine 🌞